osu.Game/Screens/Play/Player.cs

```diff
@@ -213,6 +213,12 @@ private void load(OsuConfigManager config, OsuGameBase game, CancellationToken c
             if (playableBeatmap == null)
                 return;
 
+            if (gameplayMods.Any(m => ruleset.AllMods.All(rulesetMod => m.GetType() != rulesetMod.GetType())))
+            {
+                Logger.Log($@"Gameplay was started with a mod belonging to a ruleset different than '{ruleset.Description}'.", level: LogLevel.Important);
```

Comment:
We'd hope that it doesn't get to this point since a user would have no idea how to deal with this, but I guess this is preferred to a crash.

Reply:
Hmm, a user reaching this state means that there is a clear issue that needs to be solved, so at that point it might make more sense to promote this to an exception or something that gets reported in sentry...thoughts?

Reply:
It's pretty much the same thing as

https://github.com/ppy/osu/blob/ed89f6474b42ecdc32f75e2b36171204c1fc6c83/osu.Game/Screens/Play/Player.cs#L202-L206

no?

I'm not sure there is anything to be actioned here.

Reply:
Correct, I'm proposing something new, and interested in other's thoughts on this (especially since mods of different ruleset is much worse than an unknown mod).